src/zeit/content/volume/volume.py

```diff
@@ -94,6 +90,45 @@ def _find_in_order(self, start, end, sort):
         return zeit.cms.interfaces.ICMSContent(
             iter(result).next()['uniqueId'], None)
 
+    def get_cover(self, cover_id, product_id=None, use_fallback=True):
+        path = '//covers/cover[@id="{}" and @product_id="{}"]' \
+            .format(cover_id, product_id)
+        node = self.xml.xpath(path)
+        uniqueId = node[0].get('href') if node else None
+        if uniqueId:
+            return zeit.cms.interfaces.ICMSContent(uniqueId, None)
+        # Fallback: try to find product for main product
+        # Save recursion cause product id will be equal to self.product.id
+        # in the next call
+        elif product_id != self.product.id and use_fallback:
+            return self.get_cover(cover_id, self.product.id)
+
+    def set_cover(self, cover_id, product_id, imagegroup):
+        # Check if this cover is defined in VolumeCoverSource and
+        # product is part of this volume and set it in xml
+        if self._is_valid_cover_id_and_product_id(cover_id, product_id):
+            path = '//covers/cover[@id="{}" and @product_id="{}"]' \
+                .format(cover_id, product_id)
+            node = self.xml.xpath(path)
+            if node:
+                self.xml.covers.remove(node[0])
+            if imagegroup:
```

Comment:
Fiese Details für 300: ImageGroup ist innendrin ein Folder, also hat es `__len__`, damit evaluiert `bool(imagegroup)` zu "sind da Bilder drin?". Deshalb sollte man bei Content-Typen prinzipiell `if thing is None` prüfen.